inc/saiport.h

```diff
@@ -1456,22 +1458,26 @@ typedef sai_status_t (*sai_clear_port_all_stats_fn)(
  *
  * Passed as a parameter into sai_initialize_switch()
  *
+ * @param[in] switch_id Switch id
  * @param[in] count Number of notifications
  * @param[in] data Array of port operational status
  */
 typedef void (*sai_port_state_change_notification_fn)(
```

Comment:
typedef struct _sai_port_oper_status_notification_t
{
    /** Port id */
    sai_object_id_t port_id;

    /** Port operational status */
    sai_port_oper_status_t port_state;
} sai_port_oper_status_notification_t;

This structure does not has switch id it has only port id. There is noway to get switch object id from the port object id in applications because switch object id embedding in port oid is SAI vendor specific implementation. Application will not know about format.
I am not sure is there any need to know switch id for port notifications from applications side. Since this will be uniform having switch object id for all notifications instead of retrieving from other OID's.